### PR TITLE
Add GraphQL Codegen

### PR DESCRIPTION
**Pull request by octobrekids**
In this branch, including by 
- add dependency 
> GraphQL 
> GraphQL Codegen 
> GraphQL Operation 
> GraphQL React Apollo 
> GraphQL Codegen Typescript
- add Schema / Operation GraphQL
- Generate Codegen
- Try to query mock data
- Try to use Hook that it generated from CodeGen

REMARK*
- Test is removed since it's mock data and doesn't have a necessary purpose for the test

### CODE REVIEW

**Comment by yee2542** [comment]
resolve si ka

File: src/common/tests/mock/graphql/Users.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { useQuery } from '@apollo/client';
+import { Query } from './types';
+import { GET_USER } from './query';
+
+const Users: React.FC = () => {
+  const { data, loading, error } = useQuery<Query>(GET_USER);
```
Comment (yee2542) [inline src/common/tests/mock/graphql/Users.tsx:7]: ?

File: src/common/tests/mock/graphql/mocks.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import * as Factory from 'factory.ts';
```
Comment (yee2542) [inline src/common/tests/mock/graphql/mocks.ts:1]: we have enabled `esModuleInterop` the import style no need asterik

File: src/common/tests/mock/graphql/types.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
+import { gql } from '@apollo/client';
+import * as Apollo from '@apollo/client';
```
Comment (yee2542) [inline src/common/tests/mock/graphql/types.ts:2]: same

**Review by yee2542 — CHANGES_REQUESTED** [review]
check

File: src/common/tests/mock/graphql/Users.tsx (hunk 1 above)
Comment (octobrekids) [reply src/common/tests/mock/graphql/Users.tsx]: remove unnecessary code here 81a0cc9aff630b22a39864d09a9e035b067bcfbd

File: src/common/tests/mock/graphql/mocks.ts (hunk 2 above)
Comment (octobrekids) [reply src/common/tests/mock/graphql/mocks.ts]: fix here 4c2b35fe55cbf765fc96fe1122a38fafeae7722b

File: src/common/tests/mock/graphql/types.ts (hunk 3 above)
Comment (octobrekids) [reply src/common/tests/mock/graphql/types.ts]: fix here 4c2b35fe55cbf765fc96fe1122a38fafeae7722b

File: src/common/tests/mock/graphql/mocks.ts (hunk 2 above)
Comment (octobrekids) [reply src/common/tests/mock/graphql/mocks.ts]: actually fix here e0aa3823e50675c5564185f248da32e44d18b219

**Review by yee2542 — APPROVED** [review]

**Pull request merged**